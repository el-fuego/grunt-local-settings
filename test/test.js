var grunt = require('grunt');

/*
  ======== A Handy Little Nodeunit Reference ========
  https://github.com/caolan/nodeunit

  Test methods:
    test.expect(numAssertions)
    test.done()
  Test assertions:
    test.ok(value, [message])
    test.equal(actual, expected, [message])
    test.notEqual(actual, expected, [message])
    test.deepEqual(actual, expected, [message])
    test.notDeepEqual(actual, expected, [message])
    test.strictEqual(actual, expected, [message])
    test.notStrictEqual(actual, expected, [message])
    test.throws(block, [error], [message])
    test.doesNotThrow(block, [error], [message])
    test.ifError(value)
*/

exports.templates_concat = {
    setUp: function (done) {
        'use strict';

        // setup here if necessary
        done();
    },
    testIt: function (test) {
        'use strict';
        //grunt.file.write('test/expected/test.json', JSON.stringify(grunt.config('test')));

        test.expect(1);
        var actual = grunt.file.read('test/expected/test.json');
        test.equal(actual, JSON.stringify(grunt.config('test')), 'Settings isn`t like an expected');

        test.done();
    }
};
