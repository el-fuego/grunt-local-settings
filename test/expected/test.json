{"mergedOptions":{"firstAttr":"new","secondAttr":"old","newAttr":"new"},"array":["new"],"coffeeAttr":"new","newAttr":"new","jsonAttr":"new","yamlAttr":"new"}